source/libnormaliz/vector_operations.cpp

```diff
@@ -680,8 +679,6 @@ template mpz_class v_make_prime(vector<mpz_class>&);
 template long v_scalar_product(const vector<long>& a,const vector<long>& b);
 template long long v_scalar_product(const vector<long long>& a,const vector<long long>& b);
 template mpz_class v_scalar_product(const vector<mpz_class>& a,const vector<mpz_class>& b);
-template mpq_class v_scalar_product(const vector<mpq_class>& a,const vector<mpq_class>& b);
-template nmz_float v_scalar_product(const vector<nmz_float>& a,const vector<nmz_float>& b);
```

Comment:
This templates are specialized above for the removed types, so there is no point in adding explicit instantiations of the base template spec.

@w-bruns My "fix" for the template warnings was wrong before; I just rebased this PR, let's see if the tests pass now.